pkg/azuredisk/controllerserver.go

```diff
@@ -424,6 +424,19 @@ func (d *Driver) ControllerModifyVolume(ctx context.Context, req *csi.Controller
 		skuName = ""
 	}
 
+	// Get current disk state before modification
+	currentDisk, err := d.checkDiskExists(ctx, diskURI)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("Volume not found, failed with error: %v", err))
+	}
```

Comment:
Do we really need to recheck disk existence? We are already doing at line 409, Can we reuse it?